client/a8c-for-agencies/components/items-dashboard/items-dataviews/index.tsx

```diff
@@ -78,6 +81,29 @@ const ItemsDataViews = ( { data, isLoading = false, className }: ItemsDataViewsP
 		ReactDOM.hydrate( <SpinnerWrapper />, spinnerWrapper );
 	}
 
+	useLayoutEffect( () => {
+		if (
+			! scrollContainerRef.current ||
+			previousDataViewsState?.type !== data.dataViewsState.type
+		) {
+			scrollContainerRef.current = document.querySelector(
+				'.dataviews-view-list, .dataviews-view-table-wrapper'
+			) as HTMLElement;
+		}
+
+		if ( ! previousDataViewsState?.selectedItem && data.dataViewsState.selectedItem ) {
+			window.setTimeout(
+				() => scrollContainerRef.current?.querySelector( 'li.is-selected' )?.scrollIntoView(),
+				300
+			);
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blue.svg)

Is the `setTimeout` used to ensure the DOM is updated before attempting to scroll into view? I think waiting 300ms is a reasonable approach for now!
In the future, we might consider using something like MutationObserver to handle scrolling at the optimal time without delay.

Reply:
> Is the setTimeout used to ensure the DOM is updated before attempting to scroll into view?

YES!

> In the future, we might consider using something like MutationObserver to handle scrolling at the optimal time without delay.

Do you have an idea of what should we observe 🤔

Reply:
Sorry, I haven't considered it enough yet, but perhaps we could observe `main.a4a-layout`. 
My intention was that we could revisit this if we encounter any issues with the delay. For now, it seems to be working well for me!

Reply:
In general, I don't like to use delays as a trick to fix things... but I've used it 😆  and sometimes, there are no possible alternatives...

This is another hack to fix an issue that should be fixed in the DataViews component (another one).